modules/programs/zsh.nix

```diff
@@ -190,17 +202,22 @@ in
       envVarsStr = concatStringsSep "\n" (
         mapAttrsToList export config.home.sessionVariables
       );
+
+      zdotdir = "$HOME/" + cfg.dotDir;
     in mkIf cfg.enable {
       home.packages = with pkgs; [ zsh ]
         ++ optional cfg.enableCompletion nix-zsh-completions
         ++ optional cfg.oh-my-zsh.enable oh-my-zsh;
 
+      home.sessionVariables.ZDOTDIR = if cfg.dotDir != "" then zdotdir else null;
+
       home.file.".zshenv".text = ''
         ${optionalString (config.home.sessionVariableSetter == "zsh")
           envVarsStr}
+        ${optionalString (cfg.dotDir != "") (export "ZDOTDIR" zdotdir)}
```

Comment:
I think this line should be inside the `.zshenv` file since we set session variables there. The condition could also be `cfg.dotDir != null && config.home.sessionVariableSetter != "zsh"` since if zsh is sessionVariableSetter, then it will be set automatically.

Reply:
It is in the .zshenv file though? I think this should be removed actually though, even though ZDOTDIR only applies to zsh, there is nothing really special about it for it to be in .zshenv.

Reply:
But, thinking about it again, you're right, the condition should be different and this setting should be kept in, because not ass sessionVariablesSetters apply instantly